### PR TITLE
Add sqlite feature flag

### PR DESCRIPTION
**Pull request by jackson211**
Add "sqlite" feature flag to maplibre-build-tools.
However the maplibre still needs sqlite dependency, therefore the "sqlite" feature is still needed to be enabled in maplibre. Not sure if we need another flag in maplibre?

## ✔️ PR Todo

- [x] Included links to related issues/PRs
#107

### CODE REVIEW

File: maplibre/Cargo.toml [hunk 1]
```diff
@@ -69,4 +69,4 @@ csscolorparser = { version = "0.5", features = ["serde", "cint"]}
 cint = "0.2"
 
 [build-dependencies]
-maplibre-build-tools = { path = "../maplibre-build-tools", version = "0.1.0" }
+maplibre-build-tools = { path = "../maplibre-build-tools", version = "0.1.0", features = ["sqlite"] }
```
Comment (maxammann) [inline maplibre/Cargo.toml:72]: I think we want to add a `embed-static-tiles` feature flag to `maplibre`.
That feature flag will enable the `sqlite` feature of the build-tools and conditionally embed tiles
 in the `build.rs` file.
Such that one can conditionally compile `maplibre`, once with sqlite support during building and once without.

**Review by maxammann — CHANGES_REQUESTED** [review]

**Comment by maxammann** [comment]
I updated the branch because of major refactorings on `main`.

File: maplibre-build-tools/src/mbtiles.rs [hunk 2]
```diff
@@ -83,10 +86,12 @@ pub fn extract<P: AsRef<Path>, R: AsRef<Path>>(
     Ok(())
 }
 
+#[allow(dead_code)]
 fn flip_vertical_axis(zoom: u8, value: u32) -> u32 {
```
[the remark this answers is not in the document]
Comment (jackson211) [reply maplibre-build-tools/src/mbtiles.rs]: Actually, I think we can remove dead_code flag for now as there are a lot of unfinished features.

**Comment by jackson211** [comment]
So, I have moved the `sqlite` flag from each function to the lib mod and added conditional flags in the `build.rs` file.

**Review by maxammann — APPROVED** [review]
This is fine! We can now think about removing the sqlite related code from the windows CI/CD pipeline

**Comment by maxammann** [comment]
After buids finish I'd merge this :) thank you!

**Pull request merged**